src/openrct2/rct1/S4Importer.cpp

```diff
@@ -288,33 +288,28 @@ namespace OpenRCT2::RCT1
             return {};
         }
 
+    private:
+        uint8_t calcParkValueConversionFactor(const Park::ParkData& park, const GameState_t& gameState)
+        {
+            assert(_s4.ParkValue != 0);
+
+            // Use the ratio between the old and new park value to calcute the ratio to
+            // use for the park value history and the goal.
+            return (Park::CalculateParkValue(park, gameState) * 10) / _s4.ParkValue;
+        }
+
         money64 CorrectRCT1ParkValue(money32 oldParkValue)
         {
             if (oldParkValue == kMoney32Undefined)
             {
                 return kMoney64Undefined;
             }
 
-            if (_parkValueConversionFactor == 0)
-            {
-                if (_s4.ParkValue != 0)
-                {
-                    // Use the ratio between the old and new park value to calcute the ratio to
-                    // use for the park value history and the goal.
-                    // TODO: split up this function so this can pass the actual park/gamestate as needed
-                    _parkValueConversionFactor = (Park::CalculateParkValue() * 10) / _s4.ParkValue;
-                }
-                else
-                {
-                    // In new games, the park value isn't set.
-                    _parkValueConversionFactor = 100;
-                }
-            }
+            assert(_parkValueConversionFactor != 0);
```

Comment:
Shouldn't we be more graceful in handling the parks that could be hacked? Or was it not that much prevalent in RCT1 days?

Reply:
I don’t understand. This is something we calculate ourselves based on two metadata fields. How would hacked parks affect that in any way?